src/scp/NominationProtocol.cpp

```diff
@@ -363,7 +363,23 @@ NominationProtocol::getNewValueFromNomination(SCPNomination const& nom)
                 }
             }
         }
-    });
+    };
+
+    for (auto const& val : nom.accepted)
+    {
+        pickValue(val);
+    }
+
+    // Only if we have not selected an accepted value, pick value a round leader
+    // voted for
+    if (!newVote)
```

Comment:
this is not exactly right: you don't want to introduce values from `nom.votes` if there was any `valueToNominate`. Consider the situation where `mVotes` already contains `nom.accepted`

Reply:
Oh hmm, I thought we want to introduce a value we haven't voted for either way, not no-op in this case. I can update that though, but want to make sure it's safe (so we should wait to get input from Giuliano on the issue as well)